deployment/terraform/efs.go

```diff
@@ -0,0 +1,87 @@
+package terraform
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform/ssh"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+const (
+	efsDirectory  = "/mnt/mattermost-data"
+	svcScript     = "/opt/mount.sh"
+	efcServiceDir = "/lib/systemd/system/efs-mount.service"
+	efsService    = `[Unit]
+Description=Script to run after fstab to change ownership of mount directory
+After=local-fs.target
+
+[Service]
+Type=simple
+ExecStart=/bin/bash -c "/opt/mount.sh"
+
+
+[Install]
+WantedBy=multi-user.target
+`
+)
+
+func (t *Terraform) setupEFS(extAgent *ssh.ExtAgent) error {
+	for _, val := range t.output.Instances {
+		sshc, err := extAgent.NewClient(val.PublicIP)
+		if err != nil {
+			return fmt.Errorf("error in getting ssh connection %w", err)
+		}
+		defer sshc.Close()
+
+		cmd := fmt.Sprintf("sudo mkdir -p %s", efsDirectory)
+		_, err = sshc.RunCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error creating the mounting point: %w", err)
+		}
+
+		mlog.Info("Updating /etc/fstab to mount EFS", mlog.String("fsid", t.output.EFSAccessPoint.FileSystemId))
+		// setting uid and gid to ubuntu user
+		fstabEntry := fmt.Sprintf("%s.efs.%s.amazonaws.com:/ %s nfs4 nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2,noresvport,_netdev 0 0", t.output.EFSAccessPoint.FileSystemId, t.config.AWSRegion, efsDirectory)
+		cmd = fmt.Sprintf("grep -q %q /etc/fstab || echo %q | sudo tee -a /etc/fstab > /dev/null", efsDirectory, fstabEntry)
+		_, err = sshc.RunCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error modifying /etc/fstab: %w", err)
+		}
+
+		_, err = sshc.RunCommand("sudo mount -a")
+		if err != nil {
+			return fmt.Errorf("error mounting the fstab entry: %w", err)
+		}
+
+		cmd = fmt.Sprintf("sudo chown -R ubuntu:ubuntu %s", efsDirectory)
```

Comment:
Instead of running the command, and then copying the same thing again in the svcScript, can we simply upload the script and then run the script as part of `chmod`ding it - `sudo chmod +x <file> && ./<file>`?

Reply:
We need to change mod on every reboot otherwise ownership is set to `root`. I couldn't find a better workaround for that. That's why I had to create the service.

Reply:
Yeah sure. All I am saying is instead of running the chmod command separately, we can simply execute the file.

Reply:
But do we need the script at all? Could we write `ExecStart=chown -R ubuntu:ubuntu /mnt/mattermost-data` directly in the unit file?

Reply:
Ah nice one!

Reply:
I guess we should do it in `ExecStartPre` then.